ui/lib/StreamStatus.tsx

```diff
@@ -1,36 +1,63 @@
-import * as React from "react";
+import React, { useState, useEffect } from "react";
 import { PieChart } from "react-minimal-pie-chart";
-
 import { Stream } from "@apeworx/apepay";
 
 export interface StreamStatusProps {
   stream: Stream;
+  chartType: "bar" | "pie";
 }
 
-const StreamStatus = (props: StreamStatusProps) => {
-  const [timeLeft, setTimeLeft] = React.useState(1); // using `1` to start w/ 100% left
-  React.useEffect(() => {
-    props.stream.timeLeft().then(setTimeLeft).catch(console.error);
-  }, [timeLeft]);
+const StreamStatus: React.FC<StreamStatusProps> = ({ stream, chartType }) => {
+  const [timeLeft, setTimeLeft] = useState<number>(1);
+  const [totalTime, setTotalTime] = useState<number>(1);
+
+  useEffect(() => {
+    stream.timeLeft().then(setTimeLeft).catch(console.error);
+    stream.totalTime().then(setTotalTime).catch(console.error);
+
+    // Create an interval to decrement timeLeft every second
+    const intervalId = setInterval(() => {
+      setTimeLeft((prevTimeLeft) => Math.max(prevTimeLeft - 1, 0));
+    }, 1000);
+
+    // Clear the interval when the component is unmounted
+    return () => clearInterval(intervalId);
+  }, [stream]);
+
+  console.log("totaltime " + totalTime);
+  console.log("timeleft " + timeLeft);
 
-  const [totalTime, setTotalTime] = React.useState(1); // using `1` to avoid NaN
-  React.useEffect(() => {
-    props.stream.totalTime().then(setTotalTime).catch(console.error);
-  }, [totalTime]);
+  const percentageLeft = (timeLeft / totalTime) * 100;
 
   return (
-    <PieChart
-      data={[{ value: timeLeft, color: "#111" }]}
-      totalValue={totalTime}
-      lineWidth={20}
-      background="#bfbfbf"
-      rounded
-      animate
-      label={({ dataEntry }) =>
-        `${Math.floor((100 * dataEntry.value) / totalTime)}%`
-      }
-      labelPosition={0}
-    />
+    <>
+      {chartType === "bar" ? (
+        <PieChart
+          data={[{ value: timeLeft, color: "#111" }]}
+          totalValue={totalTime}
+          lineWidth={20}
+          background="#bfbfbf"
+          rounded
+          animate
+          label={({ dataEntry }) =>
+            `${Math.floor((100 * dataEntry.value) / totalTime)}%`
```

Comment:
question: Isn't this just `percentageLeft`?

Reply:
yes nice catch